src/js/panels.js

```diff
@@ -198,16 +193,317 @@ https://github.com/fluid-project/infusion/raw/master/Infusion-LICENSE.txt
+    // the language panel. To work around this issue, another layer of containment needs to be added.
+    // This component and the demands block should be removed when the new framework (http://issues.fluidproject.org/browse/FLUID-5249)
+    // is in use.
+    fluid.defaults("gpii.firstDiscovery.panel.lang.attachTooltipOnLang", {
+        gradeNames: ["gpii.firstDiscovery.attachTooltip", "autoInit"],
+        tooltipContentMap: {
+            "prev": "navButtonTooltip",
+            "next": "navButtonTooltip"
+        }
+    });
+
+    fluid.demands("fluid.tooltip", ["gpii.firstDiscovery.panel.lang.attachTooltipOnLang"], {
+        options: {
+            styles: {
+                tooltip: "gpii-fd-tooltip-lang"
+            }
+        }
+    });
+
+    gpii.firstDiscovery.panel.lang.setNavKeyStatus = function (that) {
+        var langArray = that.options.controlValues.lang,
+            selectedLang = that.model.lang;
+
+        that.locate("prev").prop("disabled", selectedLang === langArray[0]);
+        that.locate("next").prop("disabled", selectedLang === langArray[langArray.length - 1]);
+    };
+
+    gpii.firstDiscovery.panel.lang.moveLangFocus = function (that, adjustBy) {
+        var langArray = that.options.controlValues.lang,
+            nextIndex = langArray.indexOf(that.model.lang) + adjustBy;
+
+        if (nextIndex >= 0 && nextIndex <= langArray.length) {
+            that.applier.change("lang", langArray[nextIndex]);
+        }
+    };
+
+    // find the index of the number in the "numbers" array that's closest to the given "currentNumber"
+    gpii.firstDiscovery.panel.lang.findClosestNumber = function (currentNumber, numbers) {
+        var distance = Math.abs(numbers[0] - currentNumber),
+            idx = 0;
+
+        for (var c = 1; c < numbers.length; c++) {
+            var cdistance = Math.abs(numbers[c] - currentNumber);
+            if (cdistance < distance) {
+                idx = c;
+                distance = cdistance;
+            }
+        }
+        return numbers[idx];
+    };
+
+    // When arrow keys are used to navigate thru language buttons, this function scrolls the select button
+    // to the appropriate position to ensure,
+    // 1. the selected button is in the view;
+    // 2. the top and bottom buttons are not partially shown.
+    // To achieve this, when the page is rendered, the function saves the initial positions of in-view buttons,
+    // and scroll the selected language button to the closest position. When arrow keys are used
+    // to move to an out-of-view language button into the view, also finds the closest saved position to
+    // move the button to.
+    gpii.firstDiscovery.panel.lang.scrollLangIntoView = function (that) {
+        if (!that.buttonTops) {
+            return;
+        }
+        that.lastMovedHeight = that.lastMovedHeight || 0;
```

Comment:
The use of this private variable is awkward and perhaps the biggest impediment to turning this into a generic fluid.scrollIntoView(controls, elements, index) function - is there perhaps some measurement that we can take from the DOM instead that will let us determine this? It seems that this will be fragile in the face of the user operating a scrolling control (if one appeared) in between invocations to this function